core/src/main/scala/org/apache/spark/scheduler/TaskSchedulerImpl.scala

```diff
@@ -159,6 +159,15 @@ private[spark] class TaskSchedulerImpl(
     this.synchronized {
       val manager = createTaskSetManager(taskSet, maxTaskFailures)
       activeTaskSets(taskSet.id) = manager
+      val attemptsByStage = activeTaskSets.values.map(_.taskSet).groupBy(_.stageId)
```

Comment:
Hmm... I wonder if this shouldn't be done before the new task set is added to `activeTaskSets` on the line above?

(Although I haven't really gone and tried to understand the rest of the code around this.)